components/library/uthash/patches/01-add-Makefile.patch

```diff
@@ -0,0 +1,24 @@
+--- uthash-2.3.0/Makefile.orig	2021-07-07 03:57:18.876256617 +0000
++++ uthash-2.3.0/Makefile	2021-07-07 03:57:01.242117313 +0000
+@@ -0,0 +1,21 @@
++PREFIX ?= /usr
++
++all:
++	@echo srcdir is $(srcdir)
++
++install:
++	@mkdir -p $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p utarray.h $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p uthash.h $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p utlist.h $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p utringbuffer.h $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p utstack.h $(DESTDIR)$(PREFIX)/include
++	@cd src/ && cp -p utstring.h $(DESTDIR)$(PREFIX)/include
++
++uninstall:
++	@rm -rf $(DESTDIR)$(PREFIX)/include/utarray.h
++	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash.h
```

Comment:
I think we should move the files into its own folder, eg. include/uthash/.
This is what Solaris did for many libraries.

Reply:
Would other programs be able to find the headers then? I was under the impression that `include <something.h>` searches for `something.h` in `/usr/include/`, not in any subdirectories

Reply:
Either by changing it to #include <uthash/something.h> or by adding the path to the include-search-path to gcc, eg. by adding  -I /usr/include/uthash